src/mpid/ch4/shm/xpmem/xpmem_recv.h

```diff
@@ -11,17 +11,21 @@
 #include "xpmem_pre.h"
 #include "xpmem_seg.h"
 #include "xpmem_impl.h"
+#include "xpmem_send.h"
+
+void MPIR_Datatype_iscontig(MPI_Datatype datatype, int *flag);
```

Comment:
I'm not sure why this function declaration is necessary. Shouldn't it already come from `ch4_impl.h`? If not, maybe this needs `#include "mpiimpl.h"?